File: stageTemplates/src/app/models/barrage.ts
enum LocalisationBarr {
    Amont = "Amont",
    Aval = "Aval",
    Fondation = "Fondation",
    ND = "ND",
    Noayau = "Noayau",
    RechercheAmont = "RechercheAmont",
    RechercheAval = "RechercheAval",
    RD = "RD",
    RG = "RG"
  }
  class Barrage {
    constructor(
      public id: number,
      public nom: string,
      public localisation: LocalisationBarr,
      public capacite: number,
      public villePlusProche: string,
      public hauteurBarr: number,
      public distVillePlusProche: number,
      public codeRetNormal: string,
      public dateMiseEnServ: string, // Change this to a string representation of date
      public laRetenue: string,
      public agenceId: number | null,
     
    ) {}
  }
  
  export { Barrage, LocalisationBarr};
  

File: CondorV/CondorV/Models/BD/Barrage.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace CondorV.Models.BD
{
    public enum LocalisationBarr
    {
        Amont,
        Aval,
        Fondation,
        ND,
        Noayau,
        RechercheAmont,
        RechercheAval,
        RD,
        RG,
    }
    public class Barrage
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; } 
        public string Nom { get; set; }
        public LocalisationBarr Localisation { get; set; }
        public double Capacite { get; set; }
        public string VillePlusProche { get; set; }
        public double HauteurBarr { get; set; }
        public double DistVillePlusProche { get; set; }
        public string CodeRetNormal { get; set; }
        public DateTime DateMiseEnServ { get; set; }
        public string LaRetenue { get; set; }
        //Table HSV
        //But de Barrage
        [JsonIgnore]
        public ICollection<Utilisateur>? Utilisateurs { get; set; }
        public long? AgenceId { get; set; }
        [JsonIgnore]
        public Agence? Agence { get; set; }

        public Barrage() { }
    }
}
